frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -94,14 +94,30 @@ export const StrategyForm = ({
   buttonLabels?: ButtonLabels;
   isInSidebar?: boolean;
 }) => {
-  const defaultStrategy: CacheStrategy = {
-    type: targetId === rootId ? "nocache" : "inherit",
-  };
+  const defaultStrategy: CacheStrategy = useMemo(
+    () => ({
+      type: targetId === rootId ? "nocache" : "inherit",
+    }),
+    [targetId],
+  );
+
+  // Remove default values from the form
+  const initialValues = useMemo(() => {
+    const strategy: any = { ...(savedStrategy ?? defaultStrategy) };
+    for (const fieldName in strategy) {
+      const value = strategy[fieldName as keyof CacheStrategy];
+      const defaultValue = getDefaultValueForField(strategy.type, fieldName);
+      if (defaultValue && value === defaultValue) {
+        delete strategy[fieldName as keyof CacheStrategy];
+      }
+    }
+    return strategy;
+  }, [savedStrategy, defaultStrategy]);
```

Comment:
This solution feels heavy to me. I'd love to avoid it.

One option might be to change how default values are handled in this form.

Currently, they're shown as placeholders. On master, when you delete the value and click save, the default value becomes part of the initialValues of the Formik form. But it's not displayed in the text inputs. So there's a mismatch between the saved values and the values that appear in the text inputs, and the form is considered dirty.

An alternative approach would be this: we set each text input's value to its default value (if there is one). If someone deletes the value from the input and clicks save, the default value is put into the input before the data is submitted.

Reply:
I don't think this is a good idea anymore. The conventional behavior of our forms is to use the default value as a placeholder